core/arch/arm/plat-hikey/main.c

```diff
@@ -93,3 +108,166 @@ void console_flush(void)
 {
 	pl011_flush(console_base());
 }
+
+static vaddr_t peripherals_base(paddr_t pa)
+{
+	void *va;
+
+	if (cpu_mmu_enabled()) {
+		va = phys_to_virt(pa, MEM_AREA_IO_NSEC);
+		return (vaddr_t)va;
+	}
+	return (vaddr_t)pa;
+}
+
+#ifdef CFG_SPI
+#ifdef CFG_SPI_TEST
+static void spi_test(void)
+{
+	struct pl061_data platform_pl061_data;
+	struct pl022_data platform_pl022_data;
+
+	vaddr_t gpio6_base = peripherals_base(GPIO6_BASE);
+	vaddr_t spi_base = peripherals_base(SPI_BASE);
+
+	uint8_t tx[3] = {1, 0x80, 0};
```

Comment:
To be safe, I've set loopback to true for this reference code, so the values don't mean anything in that sense, but ok, I'll add a description for what they'll do in non-loopback mode.